diff --git a/cmd/admin/overrides_add.go b/cmd/admin/overrides_add.go
index f6ccec2c3..413d33116 100644
--- a/cmd/admin/overrides_add.go
+++ b/cmd/admin/overrides_add.go
@@ -53,7 +53,11 @@ func validateIdentifiers(idents ...identifier.ACMEIdentifier) error {
 	for _, ident := range idents {
 		switch ident.Type {
 		case identifier.TypeDNS:
-			err := policy.ValidDomain(ident.Value)
+			pa, err := policy.New(map[identifier.IdentifierType]bool{"dns": true}, nil, nil)
+			if err != nil {
+				return fmt.Errorf("cannot create policy authority implementation")
+			}
+			err = pa.ValidDomain(ident.Value)
 			if err != nil {
 				return fmt.Errorf("invalid domain %s: %s", ident.Value, err)
 			}
