diff --git a/cmd/boulder-va/main.go b/cmd/boulder-va/main.go
index 032435fac..d3961512b 100644
--- a/cmd/boulder-va/main.go
+++ b/cmd/boulder-va/main.go
@@ -21,6 +21,7 @@ type Config struct {
 		RemoteVAs                   []cmd.GRPCClientConfig `validate:"omitempty,dive"`
 		MaxRemoteValidationFailures int                    `validate:"omitempty,min=0,required_with=RemoteVAs"`
 		Features                    features.Config
+		LabCADomains                []string
 	}
 
 	Syslog        cmd.SyslogConfig
@@ -115,7 +116,8 @@ func main() {
 		scope,
 		clk,
 		logger,
-		c.VA.AccountURIPrefixes)
+		c.VA.AccountURIPrefixes,
+		c.VA.LabCADomains)
 	cmd.FailOnError(err, "Unable to create VA server")
 
 	start, err := bgrpc.NewServer(c.VA.GRPC, logger).Add(
