diff --git a/cmd/boulder-va/main.go b/cmd/boulder-va/main.go
index 1c998bc97..0665bdf4b 100644
--- a/cmd/boulder-va/main.go
+++ b/cmd/boulder-va/main.go
@@ -27,7 +27,8 @@ type Config struct {
 		// before giving up. May be short-circuited by deadlines. A zero value
 		// will be turned into 1.
 		DNSTries                  int
-		DNSProvider               *cmd.DNSProvider `validate:"required"`
+		DNSResolvers              []string
+		DNSProvider               *cmd.DNSProvider `validate:"omitempty"`
 		DNSTimeout                config.Duration  `validate:"required"`
 		DNSAllowLoopbackAddresses bool
 
@@ -80,13 +81,18 @@ func main() {
 	}
 	clk := cmd.Clock()
 
-	if c.VA.DNSProvider == nil {
+	if c.VA.DNSProvider == nil && len(c.VA.DNSResolvers) == 0 {
 		cmd.Fail("Must specify dnsProvider")
 	}
 
 	var servers bdns.ServerProvider
-	servers, err = bdns.StartDynamicProvider(c.VA.DNSProvider, 60*time.Second)
-	cmd.FailOnError(err, "Couldn't start dynamic DNS server resolver")
+	if len(c.VA.DNSResolvers) > 0 {
+		servers, err = bdns.NewStaticProvider(c.VA.DNSResolvers)
+		cmd.FailOnError(err, "Couldn't parse static DNS server(s)")
+	} else {
+		servers, err = bdns.StartDynamicProvider(c.VA.DNSProvider, 60*time.Second)
+		cmd.FailOnError(err, "Couldn't start dynamic DNS server resolver")
+	}
 	defer servers.Stop()
 
 	var resolver bdns.Client
