diff --git a/cmd/boulder-ra/main.go b/cmd/boulder-ra/main.go
index 33af21a73..525410d8d 100644
--- a/cmd/boulder-ra/main.go
+++ b/cmd/boulder-ra/main.go
@@ -250,6 +250,8 @@ func main() {
 		limiterRedis, err = bredis.NewRingFromConfig(*c.RA.Limiter.Redis, scope, logger)
 		cmd.FailOnError(err, "Failed to create Redis ring")
 
+		// Set Policy Authority for ratelimits
+		ratelimits.PA = pa
 		source := ratelimits.NewRedisSource(limiterRedis.Ring, clk, scope)
 		limiter, err = ratelimits.NewLimiter(clk, source, scope)
 		cmd.FailOnError(err, "Failed to create rate limiter")
