diff --git a/cmd/boulder-va/main.go b/cmd/boulder-va/main.go
index 0bef1d4f1..ec03f44e7 100644
--- a/cmd/boulder-va/main.go
+++ b/cmd/boulder-va/main.go
@@ -41,6 +41,7 @@ type Config struct {
 		Features features.Config
 
 		AccountURIPrefixes []string `validate:"min=1,dive,required,url"`
+		LabCADomains       []string
 	}
 
 	Syslog        cmd.SyslogConfig
@@ -150,7 +151,8 @@ func main() {
 		scope,
 		clk,
 		logger,
-		c.VA.AccountURIPrefixes)
+		c.VA.AccountURIPrefixes,
+		c.VA.LabCADomains)
 	cmd.FailOnError(err, "Unable to create VA server")
 
 	start, err := bgrpc.NewServer(c.VA.GRPC, logger).Add(
