diff --git a/ca/ca.go b/ca/ca.go
index 3a62648c3..2cb8337ca 100644
--- a/ca/ca.go
+++ b/ca/ca.go
@@ -190,11 +190,12 @@ func NewCertificateAuthorityImpl(
 		}
 	}
 
-	for _, keyAlg := range []x509.PublicKeyAlgorithm{x509.ECDSA, x509.RSA} {
-		if !issuableKeys[keyAlg] {
-			fmt.Printf("WARNING: no %s issuers configured\n", keyAlg)
-		}
-	}
+	// TODO: LabCA hack!
+	//for _, keyAlg := range []x509.PublicKeyAlgorithm{x509.ECDSA, x509.RSA} {
+	//	if !issuableKeys[keyAlg] {
+	//		fmt.Printf("WARNING: no %s issuers configured\n", keyAlg)
+	//	}
+	//}
 
 	return &certificateAuthorityImpl{
 		sa:        sa,
@@ -474,9 +475,10 @@ func (ca *certificateAuthorityImpl) pickIssuer(profileName string, keyAlg x509.P
 		if !issuer.IsActive() {
 			continue
 		}
-		if issuer.KeyType() != keyAlg {
-			continue
-		}
+		// TODO: LabCA hack!
+		//if issuer.KeyType() != keyAlg {
+		//	continue
+		//}
 		if !slices.Contains(issuer.Profiles(), profileName) {
 			continue
 		}
