diff --git a/ca/crl.go b/ca/crl.go
index 99eb57b5..91e10a40 100644
--- a/ca/crl.go
+++ b/ca/crl.go
@@ -127,8 +127,10 @@ func (ci *crlImpl) GenerateCRL(stream capb.CRLGenerator_GenerateCRLServer) error
 				builder = strings.Builder{}
 			}
 		}
-		fmt.Fprint(&builder, "]")
-		ci.log.AuditInfo(builder.String())
+		if builder.Len() > 0 {
+			fmt.Fprint(&builder, "]")
+			ci.log.AuditInfo(builder.String())
+		}
 	}
 
 	template.RevokedCertificates = rcs
