diff --git a/ca/crl.go b/ca/crl.go
index 35bf4c07d..36316235e 100644
--- a/ca/crl.go
+++ b/ca/crl.go
@@ -122,8 +122,10 @@ func (ci *crlImpl) GenerateCRL(stream capb.CRLGenerator_GenerateCRLServer) error
 				builder = strings.Builder{}
 			}
 		}
-		fmt.Fprint(&builder, "]")
-		ci.log.AuditInfo(builder.String())
+		if builder.Len() > 0 {
+			fmt.Fprint(&builder, "]")
+			ci.log.AuditInfo(builder.String())
+		}
 	}
 
 	req.Entries = rcs
