diff --git a/ca/ca.go b/ca/ca.go
index 739ce53e7..2ccb11969 100644
--- a/ca/ca.go
+++ b/ca/ca.go
@@ -177,10 +177,14 @@ func makeIssuerMaps(issuers []*issuance.Issuer) (issuerMaps, error) {
 		}
 	}
 	if i, ok := issuersByAlg[x509.ECDSA]; !ok || len(i) == 0 {
-		fmt.Println("WARNING: no ECDSA issuers configured")
+		// TODO: LabCA hack!
+		issuersByAlg[x509.ECDSA] = issuersByAlg[x509.RSA]
+		// fmt.Println("WARNING: no ECDSA issuers configured")
 	}
 	if i, ok := issuersByAlg[x509.RSA]; !ok || len(i) == 0 {
-		fmt.Println("WARNING: no RSA issuers configured")
+		// TODO: LabCA hack!
+		issuersByAlg[x509.RSA] = issuersByAlg[x509.ECDSA]
+		// fmt.Println("WARNING: no RSA issuers configured")
 	}
 	return issuerMaps{issuersByAlg, issuersByNameID}, nil
 }
