diff --git a/cmd/ceremony/crl.go b/cmd/ceremony/crl.go
index 98790d906..4de35ae5c 100644
--- a/cmd/ceremony/crl.go
+++ b/cmd/ceremony/crl.go
@@ -42,7 +42,7 @@ func generateCRL(signer crypto.Signer, issuer *x509.Certificate, thisUpdate, nex
 	}
 	template.ExtraExtensions = append(template.ExtraExtensions, *idp)
 
-	err = linter.CheckCRL(template, issuer, signer, []string{})
+	err = linter.CheckCRL(template, issuer, signer, []string{"e_crl_next_update_invalid"})
 	if err != nil {
 		return nil, fmt.Errorf("crl failed pre-issuance lint: %w", err)
 	}
