diff --git a/cmd/config.go b/cmd/config.go
index 1a3edabff..09369bf88 100644
--- a/cmd/config.go
+++ b/cmd/config.go
@@ -455,7 +455,7 @@ type GRPCServerConfig struct {
 	// this controls how long it takes before a client learns about changes to its
 	// backends.
 	// https://pkg.go.dev/google.golang.org/grpc/keepalive#ServerParameters
-	MaxConnectionAge config.Duration `validate:"required"`
+	MaxConnectionAge config.Duration
 }
 
 // GRPCServiceConfig contains the information needed to configure a gRPC service.
