diff --git a/cmd/contact-auditor/main.go b/cmd/contact-auditor/main.go
index a20560b6f..ac0d567f8 100644
--- a/cmd/contact-auditor/main.go
+++ b/cmd/contact-auditor/main.go
@@ -12,6 +12,7 @@ import (
 	"time"
 
 	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
 	"github.com/letsencrypt/boulder/db"
 	blog "github.com/letsencrypt/boulder/log"
 	"github.com/letsencrypt/boulder/policy"
@@ -50,9 +51,13 @@ func validateContacts(id int64, createdAt string, contacts []string) error {
 		fmt.Fprintf(&probsBuff, "%d\t%s\tvalidation\t%q\t%q\t%q\n", id, createdAt, contact, prob, contacts)
 	}
 
+	var pa *policy.AuthorityImpl
+	logger := cmd.NewLogger(cmd.SyslogConfig{StdoutLevel: 7})
+	pa, _ = policy.New(map[core.AcmeChallenge]bool{}, logger)
+
 	for _, contact := range contacts {
 		if strings.HasPrefix(contact, "mailto:") {
-			err := policy.ValidEmail(strings.TrimPrefix(contact, "mailto:"))
+			err := pa.ValidEmail(strings.TrimPrefix(contact, "mailto:"))
 			if err != nil {
 				writeProb(contact, err.Error())
 			}
