diff --git a/cmd/crl-storer/main.go b/cmd/crl-storer/main.go
index 1f98134a..ce628793 100644
--- a/cmd/crl-storer/main.go
+++ b/cmd/crl-storer/main.go
@@ -47,6 +47,9 @@ type Config struct {
 		// https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html.
 		AWSCredsFile string
 
+		// If this is set, store the files locally instead of using (fake) S3
+		LocalStorePath string
+
 		Features map[string]bool
 	}
 
@@ -127,7 +130,7 @@ func main() {
 	}
 	s3client := s3.NewFromConfig(awsConfig, s3opts...)
 
-	csi, err := storer.New(issuers, s3client, c.CRLStorer.S3Bucket, scope, logger, clk)
+	csi, err := storer.New(issuers, s3client, c.CRLStorer.S3Bucket, c.CRLStorer.LocalStorePath, scope, logger, clk)
 	cmd.FailOnError(err, "Failed to create CRLStorer impl")
 
 	start, stop, err := bgrpc.NewServer(c.CRLStorer.GRPC).Add(
