diff --git a/cmd/crl-storer/main.go b/cmd/crl-storer/main.go
index 4dddfaa9f..8dcf40bbc 100644
--- a/cmd/crl-storer/main.go
+++ b/cmd/crl-storer/main.go
@@ -46,6 +46,9 @@ type Config struct {
 		// https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html.
 		AWSCredsFile string
 
+		// If this is set, store the files locally instead of using (fake) S3
+		LocalStorePath string
+
 		Features features.Config
 	}
 
@@ -129,7 +132,7 @@ func main() {
 	}
 	s3client := s3.NewFromConfig(awsConfig, s3opts...)
 
-	csi, err := storer.New(issuers, s3client, c.CRLStorer.S3Bucket, scope, logger, clk)
+	csi, err := storer.New(issuers, s3client, c.CRLStorer.S3Bucket, c.CRLStorer.LocalStorePath, scope, logger, clk)
 	cmd.FailOnError(err, "Failed to create CRLStorer impl")
 
 	start, err := bgrpc.NewServer(c.CRLStorer.GRPC, logger).Add(
