diff --git a/sa/db/01-incidents_sa.sql b/sa/db/01-incidents_sa.sql
index d5bea7fd5..e2b38aea2 100644
--- a/sa/db/01-incidents_sa.sql
+++ b/sa/db/01-incidents_sa.sql
@@ -1,6 +1,6 @@
 USE incidents_sa;
 
-CREATE TABLE `incident_foo` (
+CREATE TABLE IF NOT EXISTS `incident_foo` (
     `serial` varchar(255) NOT NULL,
     `registrationID` bigint(20) unsigned NULL,
     `orderID` bigint(20) unsigned NULL,
@@ -10,7 +10,7 @@ CREATE TABLE `incident_foo` (
     KEY `orderID_idx` (`orderID`)
 ) CHARSET=utf8mb4;
 
-CREATE TABLE `incident_bar` (
+CREATE TABLE IF NOT EXISTS `incident_bar` (
     `serial` varchar(255) NOT NULL,
     `registrationID` bigint(20) unsigned NULL,
     `orderID` bigint(20) unsigned NULL,
