diff --git a/sa/db/boulder_sa/20230419000000_CombinedSchema.sql b/sa/db/boulder_sa/20230419000000_CombinedSchema.sql
index 4479770fd..5c0a72435 100644
--- a/sa/db/boulder_sa/20230419000000_CombinedSchema.sql
+++ b/sa/db/boulder_sa/20230419000000_CombinedSchema.sql
@@ -1,7 +1,9 @@
 -- +migrate Up
 -- SQL in section 'Up' is executed when this migration is applied
 
-CREATE TABLE `authz2` (
+DROP TABLE IF EXISTS `authz`;
+
+CREATE TABLE IF NOT EXISTS `authz2` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `identifierType` tinyint(4) NOT NULL,
   `identifierValue` varchar(255) NOT NULL,
@@ -22,7 +24,9 @@ CREATE TABLE `authz2` (
   KEY `expires_idx` (`expires`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `blockedKeys` (
+ALTER TABLE `authz2` ADD COLUMN IF NOT EXISTS `created` datetime DEFAULT current_timestamp();
+
+CREATE TABLE IF NOT EXISTS `blockedKeys` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `keyHash` binary(32) NOT NULL,
   `added` datetime NOT NULL,
@@ -35,7 +39,7 @@ CREATE TABLE `blockedKeys` (
   KEY `extantCertificatesChecked_idx` (`extantCertificatesChecked`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `certificateStatus` (
+CREATE TABLE IF NOT EXISTS `certificateStatus` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `serial` varchar(255) NOT NULL,
   `subscriberApproved` tinyint(1) DEFAULT 0,
@@ -55,7 +59,7 @@ CREATE TABLE `certificateStatus` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `certificates` (
+CREATE TABLE IF NOT EXISTS `certificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -69,7 +73,11 @@ CREATE TABLE `certificates` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `crlShards` (
+DROP TABLE IF EXISTS `certificatesPerName`;
+DROP TABLE IF EXISTS `challenges`;
+DROP TABLE IF EXISTS `newOrdersRL`;
+
+CREATE TABLE IF NOT EXISTS `crlShards` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `issuerID` bigint(20) NOT NULL,
   `idx` int(10) unsigned NOT NULL,
@@ -80,14 +88,14 @@ CREATE TABLE `crlShards` (
   UNIQUE KEY `shardID` (`issuerID`,`idx`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `crls` (
+CREATE TABLE IF NOT EXISTS `crls` (
   `serial` varchar(255) NOT NULL,
   `createdAt` datetime NOT NULL,
   `crl` varchar(255) NOT NULL,
   PRIMARY KEY (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `fqdnSets` (
+CREATE TABLE IF NOT EXISTS `fqdnSets` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `setHash` binary(32) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -98,7 +106,7 @@ CREATE TABLE `fqdnSets` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `incidents` (
+CREATE TABLE IF NOT EXISTS `incidents` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `serialTable` varchar(128) NOT NULL,
   `url` varchar(1024) NOT NULL,
@@ -107,7 +115,7 @@ CREATE TABLE `incidents` (
   PRIMARY KEY (`id`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `issuedNames` (
+CREATE TABLE IF NOT EXISTS `issuedNames` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `reversedName` varchar(640) CHARACTER SET ascii COLLATE ascii_general_ci NOT NULL,
   `notBefore` datetime NOT NULL,
@@ -117,7 +125,7 @@ CREATE TABLE `issuedNames` (
   KEY `reversedName_notBefore_Idx` (`reversedName`,`notBefore`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `keyHashToSerial` (
+CREATE TABLE IF NOT EXISTS `keyHashToSerial` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `keyHash` binary(32) NOT NULL,
   `certNotAfter` datetime NOT NULL,
@@ -127,7 +135,7 @@ CREATE TABLE `keyHashToSerial` (
   KEY `keyHash_certNotAfter` (`keyHash`,`certNotAfter`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `orderFqdnSets` (
+CREATE TABLE IF NOT EXISTS `orderFqdnSets` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `setHash` binary(32) NOT NULL,
   `orderID` bigint(20) NOT NULL,
@@ -140,7 +148,9 @@ CREATE TABLE `orderFqdnSets` (
   KEY `orderFqdnSets_registrationID_registrations` (`registrationID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `orders` (
+ALTER TABLE `orderFqdnSets` ADD COLUMN IF NOT EXISTS `created` datetime DEFAULT current_timestamp();
+
+CREATE TABLE IF NOT EXISTS `orders` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `expires` datetime NOT NULL,
@@ -156,7 +166,12 @@ CREATE TABLE `orders` (
   KEY `regID_created_idx` (`registrationID`,`created`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `overrides` (
+ALTER TABLE `orders` ADD COLUMN IF NOT EXISTS `replaces` varchar(255) DEFAULT NULL;
+ALTER TABLE `orders` ADD COLUMN IF NOT EXISTS `authzs` blob DEFAULT NULL;
+
+DROP TABLE IF EXISTS `orderToAuthz2`;
+
+CREATE TABLE IF NOT EXISTS `overrides` (
   `limitEnum` tinyint(4) unsigned NOT NULL,
   `bucketKey` varchar(255) NOT NULL,
   `comment` varchar(255) NOT NULL,
@@ -169,7 +184,7 @@ CREATE TABLE `overrides` (
   KEY `idx_enabled` (`enabled`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `paused` (
+CREATE TABLE IF NOT EXISTS `paused` (
   `registrationID` bigint(20) unsigned NOT NULL,
   `identifierType` tinyint(4) NOT NULL,
   `identifierValue` varchar(255) NOT NULL,
@@ -181,7 +196,7 @@ CREATE TABLE `paused` (
 -- Note: This table's name is a historical artifact and it is now
 -- used to store linting certificates, not precertificates.
 -- See #6807.
-CREATE TABLE `precertificates` (
+CREATE TABLE IF NOT EXISTS `precertificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -194,7 +209,7 @@ CREATE TABLE `precertificates` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `registrations` (
+CREATE TABLE IF NOT EXISTS `registrations` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `jwk` mediumblob NOT NULL,
   `jwk_sha256` varchar(255) NOT NULL,
@@ -206,7 +221,11 @@ CREATE TABLE `registrations` (
   UNIQUE KEY `jwk_sha256` (`jwk_sha256`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `replacementOrders` (
+ALTER TABLE `registrations` DROP COLUMN IF EXISTS contact;
+ALTER TABLE `registrations` DROP COLUMN IF EXISTS initialIP;
+ALTER TABLE `registrations` MODIFY COLUMN `LockCol` bigint(20) NOT NULL DEFAULT 0;
+
+CREATE TABLE IF NOT EXISTS `replacementOrders` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `serial` varchar(255) NOT NULL,
   `orderID` bigint(20) NOT NULL,
@@ -217,7 +236,9 @@ CREATE TABLE `replacementOrders` (
   KEY `orderID_idx` (`orderID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `revokedCertificates` (
+DROP TABLE IF EXISTS `requestedNames`;
+
+CREATE TABLE IF NOT EXISTS `revokedCertificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `issuerID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -229,7 +250,9 @@ CREATE TABLE `revokedCertificates` (
   KEY `issuerID_shardIdx_notAfterHour_idx` (`issuerID`,`shardIdx`,`notAfterHour`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `serials` (
+DROP TABLE IF EXISTS `sctReceipts`;
+
+CREATE TABLE IF NOT EXISTS `serials` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -247,10 +270,38 @@ CREATE TABLE `serials` (
 -- First set of tables have foreign key constraints, so are dropped first.
 DROP TABLE `serials`;
 
+CREATE TABLE IF NOT EXISTS `authz` (
+  `id` varchar(255) NOT NULL,
+  `identifier` varchar(255) NOT NULL,
+  `registrationID` bigint(20) NOT NULL,
+  `status` varchar(255) NOT NULL,
+  `expires` datetime DEFAULT NULL,
+  `combinations` varchar(255) NOT NULL,
+  PRIMARY KEY (`id`),
+  KEY `registrationID_identifier_status_expires_authz_idx` (`registrationID`,`identifier`,`status`,`expires`),
+  CONSTRAINT `regId_authz` FOREIGN KEY (`registrationID`) REFERENCES `registrations` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
 DROP TABLE `authz2`;
 DROP TABLE `blockedKeys`;
 DROP TABLE `certificateStatus`;
 DROP TABLE `certificatesPerName`;
+
+CREATE TABLE IF NOT EXISTS `challenges` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `authorizationID` varchar(255) NOT NULL,
+  `LockCol` bigint(20) DEFAULT NULL,
+  `type` varchar(255) NOT NULL,
+  `status` varchar(255) NOT NULL,
+  `error` mediumblob DEFAULT NULL,
+  `validated` datetime DEFAULT NULL,
+  `token` varchar(255) NOT NULL,
+  `validationRecord` mediumblob DEFAULT NULL,
+  `keyAuthorization` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  KEY `authorizationID_challenges_idx` (`authorizationID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
 DROP TABLE `certificates`;
 DROP TABLE `fqdnSets`;
 DROP TABLE `incidents`;
@@ -258,8 +309,44 @@ DROP TABLE `issuedNames`;
 DROP TABLE `keyHashToSerial`;
 DROP TABLE `newOrdersRL`;
 DROP TABLE `orderFqdnSets`;
+
+CREATE TABLE IF NOT EXISTS `orderToAuthz` (
+  `orderID` bigint(20) NOT NULL,
+  `authzID` varchar(255) NOT NULL,
+  PRIMARY KEY (`orderID`,`authzID`),
+  KEY `authzID` (`authzID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
 DROP TABLE `orderToAuthz2`;
 DROP TABLE `orders`;
+
+CREATE TABLE IF NOT EXISTS `pendingAuthorizations` (
+  `id` varchar(255) NOT NULL,
+  `identifier` varchar(255) NOT NULL,
+  `registrationID` bigint(20) NOT NULL,
+  `status` varchar(255) NOT NULL,
+  `expires` datetime DEFAULT NULL,
+  `combinations` varchar(255) NOT NULL,
+  `LockCol` bigint(20) NOT NULL,
+  PRIMARY KEY (`id`),
+  KEY `identifier_registrationID_status_expires_idx` (`identifier`,`registrationID`,`status`,`expires`),
+  KEY `registrationID_status_expires_idx` (`registrationID`,`status`,`expires`),
+  CONSTRAINT `regId_pending_authz` FOREIGN KEY (`registrationID`) REFERENCES `registrations` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+CREATE TABLE IF NOT EXISTS `sctReceipts` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `sctVersion` tinyint(1) NOT NULL,
+  `logID` varchar(255) NOT NULL,
+  `timestamp` bigint(20) NOT NULL,
+  `extensions` blob DEFAULT NULL,
+  `signature` blob DEFAULT NULL,
+  `certificateSerial` varchar(255) NOT NULL,
+  `LockCol` bigint(20) DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `certificateSerial_logID` (`certificateSerial`,`logID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
 DROP TABLE `precertificates`;
 DROP TABLE `registrations`;
 DROP TABLE `requestedNames`;
