diff --git a/sa/db/boulder_sa/20230919000000_RevokedCertificates.sql b/sa/db/boulder_sa/20230919000000_RevokedCertificates.sql
index fe86aa71b..f47feb9fb 100644
--- a/sa/db/boulder_sa/20230919000000_RevokedCertificates.sql
+++ b/sa/db/boulder_sa/20230919000000_RevokedCertificates.sql
@@ -1,7 +1,7 @@
 -- +migrate Up
 -- SQL in section 'Up' is executed when this migration is applied
 
-CREATE TABLE `revokedCertificates` (
+CREATE TABLE IF NOT EXISTS `revokedCertificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `issuerID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
