diff --git a/sa/db/01-boulder_sa.sql b/sa/db/01-boulder_sa.sql
index ed3b1b39b..eb2f39225 100644
--- a/sa/db/01-boulder_sa.sql
+++ b/sa/db/01-boulder_sa.sql
@@ -1,6 +1,8 @@
 USE boulder_sa;
 
-CREATE TABLE `authz2` (
+DROP TABLE IF EXISTS `authz`;
+
+CREATE TABLE IF NOT EXISTS `authz2` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `identifierType` tinyint(4) NOT NULL,
   `identifierValue` varchar(255) NOT NULL,
@@ -21,7 +23,9 @@ CREATE TABLE `authz2` (
   KEY `expires_idx` (`expires`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `blockedKeys` (
+ALTER TABLE `authz2` ADD COLUMN IF NOT EXISTS `created` datetime DEFAULT current_timestamp();
+
+CREATE TABLE IF NOT EXISTS `blockedKeys` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `keyHash` binary(32) NOT NULL,
   `added` datetime NOT NULL,
@@ -34,7 +38,7 @@ CREATE TABLE `blockedKeys` (
   KEY `extantCertificatesChecked_idx` (`extantCertificatesChecked`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `certificateStatus` (
+CREATE TABLE IF NOT EXISTS `certificateStatus` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `serial` varchar(255) NOT NULL,
   `subscriberApproved` tinyint(1) DEFAULT 0,
@@ -54,7 +58,7 @@ CREATE TABLE `certificateStatus` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `certificates` (
+CREATE TABLE IF NOT EXISTS `certificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -68,7 +72,11 @@ CREATE TABLE `certificates` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `crlShards` (
+DROP TABLE IF EXISTS `certificatesPerName`;
+DROP TABLE IF EXISTS `challenges`;
+DROP TABLE IF EXISTS `newOrdersRL`;
+
+CREATE TABLE IF NOT EXISTS `crlShards` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `issuerID` bigint(20) NOT NULL,
   `idx` int(10) unsigned NOT NULL,
@@ -79,14 +87,14 @@ CREATE TABLE `crlShards` (
   UNIQUE KEY `shardID` (`issuerID`,`idx`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `crls` (
+CREATE TABLE IF NOT EXISTS `crls` (
   `serial` varchar(255) NOT NULL,
   `createdAt` datetime NOT NULL,
   `crl` varchar(255) NOT NULL,
   PRIMARY KEY (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `fqdnSets` (
+CREATE TABLE IF NOT EXISTS `fqdnSets` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `setHash` binary(32) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -97,7 +105,7 @@ CREATE TABLE `fqdnSets` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `incidents` (
+CREATE TABLE IF NOT EXISTS `incidents` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `serialTable` varchar(128) NOT NULL,
   `url` varchar(1024) NOT NULL,
@@ -106,7 +114,7 @@ CREATE TABLE `incidents` (
   PRIMARY KEY (`id`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `issuedNames` (
+CREATE TABLE IF NOT EXISTS `issuedNames` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `reversedName` varchar(640) CHARACTER SET ascii COLLATE ascii_general_ci NOT NULL,
   `notBefore` datetime NOT NULL,
@@ -116,7 +124,7 @@ CREATE TABLE `issuedNames` (
   KEY `reversedName_notBefore_Idx` (`reversedName`,`notBefore`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `keyHashToSerial` (
+CREATE TABLE IF NOT EXISTS `keyHashToSerial` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `keyHash` binary(32) NOT NULL,
   `certNotAfter` datetime NOT NULL,
@@ -126,7 +134,7 @@ CREATE TABLE `keyHashToSerial` (
   KEY `keyHash_certNotAfter` (`keyHash`,`certNotAfter`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `orderFqdnSets` (
+CREATE TABLE IF NOT EXISTS `orderFqdnSets` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `setHash` binary(32) NOT NULL,
   `orderID` bigint(20) NOT NULL,
@@ -139,7 +147,9 @@ CREATE TABLE `orderFqdnSets` (
   KEY `orderFqdnSets_registrationID_registrations` (`registrationID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `orders` (
+ALTER TABLE `orderFqdnSets` ADD COLUMN IF NOT EXISTS `created` datetime DEFAULT current_timestamp();
+
+CREATE TABLE IF NOT EXISTS `orders` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `expires` datetime NOT NULL,
@@ -157,7 +167,12 @@ CREATE TABLE `orders` (
   KEY `regID_created_idx` (`registrationID`,`created`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `overrides` (
+ALTER TABLE `orders` ADD COLUMN IF NOT EXISTS `replaces` varchar(255) DEFAULT NULL;
+ALTER TABLE `orders` ADD COLUMN IF NOT EXISTS `authzs` blob DEFAULT NULL;
+
+DROP TABLE IF EXISTS `orderToAuthz2`;
+
+CREATE TABLE IF NOT EXISTS `overrides` (
   `limitEnum` tinyint(4) unsigned NOT NULL,
   `bucketKey` varchar(255) NOT NULL,
   `comment` varchar(255) NOT NULL,
@@ -170,7 +185,7 @@ CREATE TABLE `overrides` (
   KEY `idx_enabled` (`enabled`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `paused` (
+CREATE TABLE IF NOT EXISTS `paused` (
   `registrationID` bigint(20) unsigned NOT NULL,
   `identifierType` tinyint(4) NOT NULL,
   `identifierValue` varchar(255) NOT NULL,
@@ -182,7 +197,7 @@ CREATE TABLE `paused` (
 -- Note: This table's name is a historical artifact and it is now
 -- used to store linting certificates, not precertificates.
 -- See #6807.
-CREATE TABLE `precertificates` (
+CREATE TABLE IF NOT EXISTS `precertificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -195,7 +210,7 @@ CREATE TABLE `precertificates` (
   KEY `serial` (`serial`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `registrations` (
+CREATE TABLE IF NOT EXISTS `registrations` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `jwk` mediumblob NOT NULL,
   `jwk_sha256` varchar(255) NOT NULL,
@@ -206,7 +221,11 @@ CREATE TABLE `registrations` (
   UNIQUE KEY `jwk_sha256` (`jwk_sha256`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci;
 
-CREATE TABLE `replacementOrders` (
+ALTER TABLE `registrations` DROP COLUMN IF EXISTS contact;
+ALTER TABLE `registrations` DROP COLUMN IF EXISTS initialIP;
+ALTER TABLE `registrations` MODIFY COLUMN IF EXISTS `LockCol` bigint(20) NOT NULL DEFAULT 0;
+
+CREATE TABLE IF NOT EXISTS `replacementOrders` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `serial` varchar(255) NOT NULL,
   `orderID` bigint(20) NOT NULL,
@@ -217,7 +236,9 @@ CREATE TABLE `replacementOrders` (
   KEY `orderID_idx` (`orderID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `revokedCertificates` (
+DROP TABLE IF EXISTS `requestedNames`;
+
+CREATE TABLE IF NOT EXISTS `revokedCertificates` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `issuerID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
@@ -229,7 +250,9 @@ CREATE TABLE `revokedCertificates` (
   KEY `issuerID_shardIdx_notAfterHour_idx` (`issuerID`,`shardIdx`,`notAfterHour`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
 
-CREATE TABLE `serials` (
+DROP TABLE IF EXISTS `sctReceipts`;
+
+CREATE TABLE IF NOT EXISTS `serials` (
   `id` bigint(20) NOT NULL AUTO_INCREMENT,
   `registrationID` bigint(20) NOT NULL,
   `serial` varchar(255) NOT NULL,
