diff --git a/test/entrypoint.sh b/test/entrypoint.sh
index a47fd2c9a..626d57155 100755
--- a/test/entrypoint.sh
+++ b/test/entrypoint.sh
@@ -13,15 +13,15 @@ service rsyslog start
 # make sure we can reach the mysqldb.
 ./test/wait-for-it.sh boulder-mysql 3306
 
-# make sure we can reach the proxysql.
-./test/wait-for-it.sh bproxysql 6032
-
 # make sure we can reach pkilint
 ./test/wait-for-it.sh bpkilint 80
 
 # create the database
 MYSQL_CONTAINER=1 $DIR/create_db.sh
 
+# Generate the internal keys and certs
+./test/certs/generate.sh
+
 if [[ $# -eq 0 ]]; then
     exec python3 ./start.py
 fi
