diff --git a/test/entrypoint.sh b/test/entrypoint.sh
index eb158c3d9..1fbfccc94 100755
--- a/test/entrypoint.sh
+++ b/test/entrypoint.sh
@@ -40,12 +40,8 @@ configure_database_endpoints() {
 # Defaults to MariaDB/ProxySQL unless USE_VITESS is true.
 configure_database_endpoints
 
-# make sure we can reach mariadb and proxysql
-./test/wait-for-it.sh boulder-mariadb 3306
-./test/wait-for-it.sh boulder-proxysql 6033
-
-# make sure we can reach vitess
-./test/wait-for-it.sh boulder-vitess 33577
+# make sure we can reach mysql
+./test/wait-for-it.sh boulder-mysql 3306
 
 # make sure we can reach pkilint
 ./test/wait-for-it.sh bpkimetal 8080
@@ -58,7 +54,7 @@ if [[ "${USE_VITESS}" == "true" ]]; then
   SKIP_USERS=1 \
   "$DIR/create_db.sh"
 else
-  DB_HOST="boulder-mariadb" \
+  DB_HOST="boulder-mysql" \
   DB_PORT=3306 \
   DB_CONFIG_FILE="${DIR}/../sa/db/dbconfig.mariadb.yml" \
   SKIP_CREATE=0 \
@@ -66,6 +62,9 @@ else
   "$DIR/create_db.sh"
 fi
 
+# Generate the internal keys and certs
+./test/certs/generate.sh
+
 if [[ $# -eq 0 ]]; then
     exec python3 ./start.py
 fi
