diff --git a/test/entrypoint.sh b/test/entrypoint.sh
index bf5301e79..783473ef8 100755
--- a/test/entrypoint.sh
+++ b/test/entrypoint.sh
@@ -22,7 +22,7 @@ DB_URL_FILES=(
 
 configure_database_endpoints() {
   DB_STYLE="proxysql"
-  export DB_ADDR="boulder-proxysql:6033"
+  export DB_ADDR="boulder-mysql:3306"
 
   if [[ "${USE_VITESS}" == "true" ]]
   then
@@ -43,13 +43,18 @@ configure_database_endpoints() {
 # Defaults to MariaDB/ProxySQL unless USE_VITESS is true.
 configure_database_endpoints
 
-# make sure we can reach mariadb and proxysql
-./test/wait-for-it.sh boulder-mariadb 3306
-./test/wait-for-it.sh boulder-proxysql 6033
+# make sure we can reach mysql
+./test/wait-for-it.sh boulder-mysql 3306
 
 # make sure pkimetal's unix socket is ready
 ./test/wait-for-socket.sh /var/run/pkimetal/pkimetal.sock
 
+# Create the database if not existing
+"$DIR/create_db.sh"
+
+# Generate the internal keys and certs
+./test/certs/generate.sh
+
 if [[ $# -eq 0 ]]; then
     exec python3 ./start.py
 fi
