diff --git a/errors/errors.go b/errors/errors.go
index 206857bd..9b185064 100644
--- a/errors/errors.go
+++ b/errors/errors.go
@@ -168,10 +168,10 @@ func NotFoundError(msg string, args ...interface{}) error {
 	return New(NotFound, msg, args...)
 }
 
-func RateLimitError(retryAfter time.Duration, msg string, args ...interface{}) error {
+func RateLimitError(errURL string, retryAfter time.Duration, msg string, args ...interface{}) error {
 	return &BoulderError{
 		Type:       RateLimit,
-		Detail:     fmt.Sprintf(msg+": see https://letsencrypt.org/docs/rate-limits/", args...),
+		Detail:     fmt.Sprintf(msg+": see "+errURL, args...),
 		RetryAfter: retryAfter,
 	}
 }
