diff --git a/cmd/expiration-mailer/main.go b/cmd/expiration-mailer/main.go
index d41bc5a0b..cde94e5a0 100644
--- a/cmd/expiration-mailer/main.go
+++ b/cmd/expiration-mailer/main.go
@@ -23,6 +23,7 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	"github.com/letsencrypt/boulder/bdns"
 	"github.com/letsencrypt/boulder/cmd"
 	"github.com/letsencrypt/boulder/config"
 	"github.com/letsencrypt/boulder/core"
@@ -38,7 +39,7 @@ import (
 )
 
 const (
-	defaultExpirationSubject = "Let's Encrypt certificate expiration notice for domain {{.ExpirationSubject}}"
+	defaultExpirationSubject = "LabCA certificate expiration notice for domain {{.ExpirationSubject}}"
 )
 
 type regStore interface {
@@ -685,6 +686,11 @@ type Config struct {
 		TLS       cmd.TLSConfig
 		SAService *cmd.GRPCClientConfig
 
+		DNSTries                  int
+		DNSResolvers              []string
+		DNSTimeout                string
+		DNSAllowLoopbackAddresses bool
+
 		// Path to a file containing a list of trusted root certificates for use
 		// during the SMTP connection (as opposed to the gRPC connections).
 		SMTPTrustedRootFile string
@@ -833,6 +839,29 @@ func main() {
 	cmd.FailOnError(err, "Failed to load credentials and create gRPC connection to SA")
 	sac := sapb.NewStorageAuthorityClient(conn)
 
+	dnsTimeout, err := time.ParseDuration(c.Mailer.DNSTimeout)
+	cmd.FailOnError(err, "Couldn't parse DNS timeout")
+	dnsTries := c.Mailer.DNSTries
+	if dnsTries < 1 {
+		dnsTries = 1
+	}
+	var resolver bdns.Client
+	servers, err := bdns.NewStaticProvider(c.Mailer.DNSResolvers)
+	cmd.FailOnError(err, "Couldn't parse static DNS server(s)")
+	if !c.Mailer.DNSAllowLoopbackAddresses {
+		r := bdns.New(
+			dnsTimeout,
+			servers,
+			scope,
+			clk,
+			dnsTries,
+			logger)
+		resolver = r
+	} else {
+		r := bdns.NewTest(dnsTimeout, servers, scope, clk, dnsTries, logger)
+		resolver = r
+	}
+
 	var smtpRoots *x509.CertPool
 	if c.Mailer.SMTPTrustedRootFile != "" {
 		pem, err := os.ReadFile(c.Mailer.SMTPTrustedRootFile)
@@ -868,6 +897,7 @@ func main() {
 		c.Mailer.Username,
 		smtpPassword,
 		smtpRoots,
+		resolver,
 		*fromAddress,
 		logger,
 		scope,
