diff --git a/features/features.go b/features/features.go
index 42dd6b7eb..ed8cbe490 100644
--- a/features/features.go
+++ b/features/features.go
@@ -15,6 +15,9 @@ import (
 // then call features.Set(parsedConfig) to load the parsed struct into this
 // package's global Config.
 type Config struct {
+	// flags for LabCA
+	DOH bool
+
 	// CertCheckerChecksValidations enables an extra query for each certificate
 	// checked, to find the relevant authzs. Since this query might be
 	// expensive, we gate it behind a feature flag.
