diff --git a/issuance/issuer.go b/issuance/issuer.go
index 95d2f03a7..c3129fe97 100644
--- a/issuance/issuer.go
+++ b/issuance/issuer.go
@@ -161,7 +161,7 @@ type IssuerConfig struct {
 	Active bool
 
 	IssuerURL  string `validate:"required,url"`
-	CRLURLBase string `validate:"required,url,startswith=http://,endswith=/"`
+	CRLURLBase string `validate:"required,url,startswith=http://"`
 
 	// TODO(#8177): Remove this.
 	OCSPURL string `validate:"omitempty,url"`
@@ -248,9 +248,6 @@ func newIssuer(config IssuerConfig, cert *Certificate, signer crypto.Signer, clk
 	if !strings.HasPrefix(config.CRLURLBase, "http://") {
 		return nil, fmt.Errorf("crlURLBase must use HTTP scheme, got %q", config.CRLURLBase)
 	}
-	if !strings.HasSuffix(config.CRLURLBase, "/") {
-		return nil, fmt.Errorf("crlURLBase must end with exactly one forward slash, got %q", config.CRLURLBase)
-	}
 
 	// We require that all of our issuers be capable of both issuing certs and
 	// providing revocation information.
