diff --git a/log/log.go b/log/log.go
index 02368437..dc39bcdd 100644
--- a/log/log.go
+++ b/log/log.go
@@ -93,6 +93,9 @@ func newStdoutWriter(level int) *stdoutWriter {
 	}
 
 	prefix := fmt.Sprintf("%s %s %s[%d]:", shortHostname, datacenter, path.Base(os.Args[0]), os.Getpid())
+	if datacenter == "unknown" {
+		prefix = fmt.Sprintf("%s %s[%d]:", shortHostname, path.Base(os.Args[0]), os.Getpid())
+	}
 
 	return &stdoutWriter{
 		prefix: prefix,
