diff --git a/log/validator/validator.go b/log/validator/validator.go
index 6b02f83ae..4b066b44a 100644
--- a/log/validator/validator.go
+++ b/log/validator/validator.go
@@ -203,9 +203,9 @@ func lineValid(text string) error {
 	if strings.Contains(text, errorPrefix) {
 		return nil
 	}
-	// Check the extracted checksum against the computed checksum
+	// Check the extracted checksum against the computed checksum, but ignore "message repeated X times" lines
 	computedChecksum := log.LogLineChecksum(line)
-	if checksum != computedChecksum {
+	if checksum != computedChecksum && checksum != "message" {
 		return fmt.Errorf("%s invalid checksum (expected %q, got %q)", errorPrefix, computedChecksum, checksum)
 	}
 	return nil
