diff --git a/cmd/notify-mailer/main.go b/cmd/notify-mailer/main.go
index f281d85c..449845dc 100644
--- a/cmd/notify-mailer/main.go
+++ b/cmd/notify-mailer/main.go
@@ -36,6 +36,7 @@ type mailer struct {
 	recipients    []recipient
 	targetRange   interval
 	sleepInterval time.Duration
+	pa            *policy.AuthorityImpl
 }
 
 // interval defines a range of email addresses to send to in alphabetical order.
@@ -171,7 +172,7 @@ func (m *mailer) run() error {
 			continue
 		}
 
-		err := policy.ValidEmail(address)
+		err := m.pa.ValidEmail(address)
 		if err != nil {
 			m.log.Infof("Skipping %q due to policy violation: %s", address, err)
 			continue
@@ -472,7 +473,9 @@ type Config struct {
 	NotifyMailer struct {
 		DB cmd.DBConfig
 		cmd.SMTPConfig
+		cmd.HostnamePolicyConfig
 	}
+	PA cmd.PAConfig
 	Syslog cmd.SyslogConfig
 }
 
@@ -536,6 +539,14 @@ func main() {
 		log.Infof("While reading the recipient list file %s", probs)
 	}
 
+	// Validate PA config and set defaults if needed
+	cmd.FailOnError(cfg.PA.CheckChallenges(), "Invalid PA configuration")
+
+	pa, err := policy.New(cfg.PA.Challenges)
+	cmd.FailOnError(err, "Failed to create PA")
+	err = pa.SetHostnamePolicyFile(cfg.NotifyMailer.HostnamePolicyFile)
+	cmd.FailOnError(err, "Failed to load HostnamePolicyFile")
+
 	var mailClient bmail.Mailer
 	if *dryRun {
 		log.Infof("Starting %s in dry-run mode", cmd.VersionString())
@@ -551,6 +562,7 @@ func main() {
 			cfg.NotifyMailer.Username,
 			smtpPassword,
 			nil,
+			nil,
 			*address,
 			log,
 			metrics.NoopRegisterer,
@@ -571,6 +583,7 @@ func main() {
 			end:   *end,
 		},
 		sleepInterval: *sleep,
+		pa:            pa,
 	}
 
 	err = m.run()
