diff --git a/ra/ra.go b/ra/ra.go
index 6daa05506..6ffd6b5df 100644
--- a/ra/ra.go
+++ b/ra/ra.go
@@ -1773,6 +1773,9 @@ func crlShard(cert *x509.Certificate) (int64, error) {
 		return 0, fmt.Errorf("malformed CRLDistributionPoint %q", url)
 	}
 	shardStr := url[lastIndex+1:]
+	if strings.HasSuffix(shardStr, "-crl.pem") {
+		return 1, nil
+	}
 	shardIdx, err := strconv.Atoi(shardStr)
 	if err != nil {
 		return 0, fmt.Errorf("parsing CRLDistributionPoint: %s", err)
