diff --git a/ra/ra.go b/ra/ra.go
index 091a40ab6..a89f1e3e2 100644
--- a/ra/ra.go
+++ b/ra/ra.go
@@ -43,7 +43,6 @@ import (
 	"github.com/letsencrypt/boulder/issuance"
 	blog "github.com/letsencrypt/boulder/log"
 	"github.com/letsencrypt/boulder/metrics"
-	"github.com/letsencrypt/boulder/policy"
 	"github.com/letsencrypt/boulder/probs"
 	pubpb "github.com/letsencrypt/boulder/publisher/proto"
 	rapb "github.com/letsencrypt/boulder/ra/proto"
@@ -593,7 +592,7 @@ func (ra *RegistrationAuthorityImpl) validateContacts(contacts []string) error {
 		if !core.IsASCII(contact) {
 			return berrors.InvalidEmailError("contact email contains non-ASCII characters")
 		}
-		err = policy.ValidEmail(parsed.Opaque)
+		err = ra.PA.ValidEmail(parsed.Opaque)
 		if err != nil {
 			return err
 		}
@@ -1906,6 +1905,9 @@ func crlShard(cert *x509.Certificate) (int64, error) {
 		return 0, fmt.Errorf("malformed CRLDistributionPoint %q", url)
 	}
 	shardStr := url[lastIndex+1:]
+	if strings.HasSuffix(shardStr, "-crl.pem") {
+		return 1, nil
+	}
 	shardIdx, err := strconv.Atoi(shardStr)
 	if err != nil {
 		return 0, fmt.Errorf("parsing CRLDistributionPoint: %s", err)
