diff --git a/ratelimit/rate-limits.go b/ratelimit/rate-limits.go
index 0d52801d..bd451521 100644
--- a/ratelimit/rate-limits.go
+++ b/ratelimit/rate-limits.go
@@ -57,6 +57,7 @@ type Limits interface {
 	CertificatesPerFQDNSetFast() RateLimitPolicy
 	NewOrdersPerAccount() RateLimitPolicy
 	LoadPolicies(contents []byte) error
+	RateLimitsURL() string
 }
 
 // limitsImpl is an unexported implementation of the Limits interface. It acts
@@ -140,6 +141,15 @@ func (r *limitsImpl) NewOrdersPerAccount() RateLimitPolicy {
 	return r.rlPolicy.NewOrdersPerAccount
 }
 
+func (r *limitsImpl) RateLimitsURL() string {
+	r.RLock()
+	defer r.RUnlock()
+	if r.rlPolicy == nil {
+		return ""
+	}
+	return r.rlPolicy.RateLimitsURL
+}
+
 // LoadPolicies loads various rate limiting policies from a byte array of
 // YAML configuration (typically read from disk by a reloader)
 func (r *limitsImpl) LoadPolicies(contents []byte) error {
@@ -194,6 +204,8 @@ type rateLimitConfig struct {
 	// lower threshold and smaller window), so that clients don't have to wait
 	// a long time after a small burst of accidental duplicate issuance.
 	CertificatesPerFQDNSetFast RateLimitPolicy `yaml:"certificatesPerFQDNSetFast"`
+	// URL to show in error messages when a rate-limit error is shown
+	RateLimitsURL string `yaml:"rateLimitsURL"`
 }
 
 // RateLimitPolicy describes a general limiting policy
