diff --git a/ratelimits/names.go b/ratelimits/names.go
index b0d581e76..a4774ae10 100644
--- a/ratelimits/names.go
+++ b/ratelimits/names.go
@@ -135,7 +135,11 @@ func validateRegId(id string) error {
 // validateDomain validates that the provided string is formatted 'domain',
 // where domain is a domain name.
 func validateDomain(id string) error {
-	err := policy.ValidDomain(id)
+	pa, err := policy.New(nil, nil)
+	if err != nil {
+		return fmt.Errorf("cannot create policy authority implementation")
+	}
+	err = pa.ValidDomain(id)
 	if err != nil {
 		return fmt.Errorf("invalid domain, %q must be formatted 'domain'", id)
 	}
@@ -150,8 +154,12 @@ func validateFQDNSet(id string) error {
 		return fmt.Errorf(
 			"invalid fqdnSet, %q must be formatted 'fqdnSet'", id)
 	}
+	pa, err := policy.New(nil, nil)
+	if err != nil {
+		return fmt.Errorf("cannot create policy authority implementation")
+	}
 	for _, domain := range domains {
-		err := policy.ValidDomain(domain)
+		err = pa.ValidDomain(domain)
 		if err != nil {
 			return fmt.Errorf(
 				"invalid domain, %q must be formatted 'fqdnSet'", id)
