diff --git a/reloader/reloader.go b/reloader/reloader.go
index 7200d707a..989db0eec 100644
--- a/reloader/reloader.go
+++ b/reloader/reloader.go
@@ -12,7 +12,7 @@ import (
 
 // Wrap time.Tick so we can override it in tests.
 var makeTicker = func() (func(), <-chan time.Time) {
-	t := time.NewTicker(1 * time.Second)
+	t := time.NewTicker(30 * time.Second)
 	return t.Stop, t.C
 }
 
@@ -56,8 +56,12 @@ func New(filename string, dataCallback func([]byte) error, logger blog.Logger) (
 			case <-tickChan:
 				currentFileInfo, err := statFile(filename)
 				if err != nil {
-					logger.Errf("statting %s: %s", filename, err)
-					continue
+					time.Sleep(10 * time.Second)
+					currentFileInfo, err = statFile(filename)
+					if err != nil {
+						logger.Errf("statting %s: %s", filename, err)
+						continue
+					}
 				}
 				if !currentFileInfo.ModTime().After(fileInfo.ModTime()) {
 					continue
