diff --git a/cmd/remoteva/main.go b/cmd/remoteva/main.go
index f99ded497..9a1033a87 100644
--- a/cmd/remoteva/main.go
+++ b/cmd/remoteva/main.go
@@ -56,7 +56,8 @@ type Config struct {
 		// For more information, see: https://pkg.go.dev/crypto/tls#ClientAuthType
 		SkipGRPCClientCertVerification bool
 
-		Features features.Config
+		Features     features.Config
+		LabCADomains []string
 	}
 
 	Syslog        cmd.SyslogConfig
@@ -141,7 +142,8 @@ func main() {
 		c.RVA.AccountURIPrefixes,
 		c.RVA.Perspective,
 		c.RVA.RIR,
-		bdns.IsReservedIP)
+		bdns.IsReservedIP,
+		c.RVA.LabCADomains)
 	cmd.FailOnError(err, "Unable to create Remote-VA server")
 
 	start, err := bgrpc.NewServer(c.RVA.GRPC, logger).Add(
