diff --git a/test/startservers.py b/test/startservers.py
index b5fc73439..135e6c9e4 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -146,6 +146,9 @@ processes = []
 challSrvProcess = None
 
 def setupHierarchy():
+    pass
+
+def setupHierarchyOriginal():
     """Set up the issuance hierarchy. Must have called install() before this."""
     e = os.environ.copy()
     e.setdefault("GOBIN", "%s/bin" % os.getcwd())
