diff --git a/test/startservers.py b/test/startservers.py
index fcfdc9423..5d19996ad 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -169,6 +169,9 @@ processes = []
 challSrvProcess = None
 
 def setupHierarchy():
+    pass
+
+def setupHierarchyOriginal():
     """Set up the issuance hierarchy. Must have called install() before this."""
     e = os.environ.copy()
     e.setdefault("GOBIN", "%s/bin" % os.getcwd())
