diff --git a/test/startservers.py b/test/startservers.py
index 0df56a4..14aa548 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -142,6 +142,9 @@ processes = []
 challSrvProcess = None
 
 def setupHierarchy():
+    pass
+
+def setupHierarchyOriginal():
     """Set up the issuance hierarchy. Must have called install() before this."""
     e = os.environ.copy()
     e.setdefault("GOBIN", "%s/bin" % os.getcwd())
