diff --git a/test/startservers.py b/test/startservers.py
index a85339777..10fd2047e 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -145,6 +145,9 @@ processes = []
 challSrvProcess = None
 
 def setupHierarchy():
+    pass
+
+def setupHierarchyOriginal():
     """Set up the issuance hierarchy. Must have called install() before this."""
     e = os.environ.copy()
     e.setdefault("GOBIN", "%s/bin" % os.getcwd())
