diff --git a/test/health-checker/main.go b/test/health-checker/main.go
index addb669bb..119867d6f 100644
--- a/test/health-checker/main.go
+++ b/test/health-checker/main.go
@@ -57,7 +57,7 @@ func main() {
 	clk := clock.New()
 
 	// Health check retry and timeout.
-	ticker := time.NewTicker(100 * time.Millisecond)
+	ticker := time.NewTicker(1 * time.Second)
 	ctx, cancel := context.WithTimeout(context.Background(), 10*c.GRPC.Timeout.Duration)
 	defer cancel()
 
