diff --git a/test/ocsp/helper/helper.go b/test/ocsp/helper/helper.go
index 469c8cec1..0b2852330 100644
--- a/test/ocsp/helper/helper.go
+++ b/test/ocsp/helper/helper.go
@@ -15,6 +15,7 @@ import (
 	"net/http"
 	"net/url"
 	"os"
+	"path"
 	"strings"
 	"sync"
 	"time"
@@ -327,7 +328,7 @@ func sendHTTPRequest(
 	var httpRequest *http.Request
 	var err error
 	if method == "GET" {
-		ocspURL.Path = encodedReq
+		ocspURL.Path = path.Join(ocspURL.Path, encodedReq)
 		fmt.Fprintf(output, "Fetching %s\n", ocspURL.String())
 		httpRequest, err = http.NewRequest("GET", ocspURL.String(), http.NoBody)
 	} else if method == "POST" {
