diff --git a/test/certs/generate.sh b/test/certs/generate.sh
index 7386e7992..50ec741f3 100755
--- a/test/certs/generate.sh
+++ b/test/certs/generate.sh
@@ -63,9 +63,9 @@ webpki() (
   # This function executes in a subshell, so this cd does not affect the parent
   # script.
   cd ../..
-  make build
-  mkdir ./test/certs/webpki
-  go run ./test/certs/webpki.go
+  # make build
+  mkdir ./labca/certs/webpki
+  # go run ./labca/certs/webpki.go
 )
 
 if ! [ -d ipki ]; then
@@ -73,6 +73,17 @@ if ! [ -d ipki ]; then
   ipki
 fi
 
+end_date=$(openssl x509 -enddate -noout -in ipki/boulder/cert.pem | cut -d= -f2)
+end_date_seconds=$(date -d "$end_date" +%s)
+current_date_seconds=$(date +%s)
+remaining_days=$(( (end_date_seconds - current_date_seconds) / 86400 ))
+if [ $remaining_days -lt 60 ]; then
+  echo "Regenerating ipki/..."
+  [ -d ipki.prev ] && rm -rf ipki.prev
+  mv ipki ipki.prev
+  ipki
+fi
+
 if ! [ -d webpki ]; then
   echo "Generating webpki/..."
   webpki
