diff --git a/crl/updater/updater.go b/crl/updater/updater.go
index fec242794..ecda37738 100644
--- a/crl/updater/updater.go
+++ b/crl/updater/updater.go
@@ -231,7 +231,7 @@ func (cu *crlUpdater) updateShard(ctx context.Context, atTime time.Time, issuerN
 			crlEntries = append(crlEntries, entry)
 		}
 
-		cu.log.Infof(
+		cu.log.Debugf(
 			"Queried SA for CRL shard: id=[%s] expiresAfter=[%s] expiresBefore=[%s] numEntries=[%d]",
 			crlID, chunk.start, chunk.end, len(crlEntries))
 	}
