diff --git a/crl/updater/updater.go b/crl/updater/updater.go
index 4d5b06b38..d7cc6dba3 100644
--- a/crl/updater/updater.go
+++ b/crl/updater/updater.go
@@ -307,7 +307,7 @@ func (cu *crlUpdater) updateShard(ctx context.Context, atTime time.Time, issuerN
 			return fmt.Errorf("streaming GetRevokedCerts: %w", err)
 		}
 
-		cu.log.Infof(
+		cu.log.Debugf(
 			"Queried SA for CRL shard: id=[%s] expiresAfter=[%s] expiresBefore=[%s] numEntries=[%d]",
 			crlID, chunk.start, chunk.end, n)
 	}
