diff --git a/va/http.go b/va/http.go
index 00942ede3..2b4ece730 100644
--- a/va/http.go
+++ b/va/http.go
@@ -341,7 +341,16 @@ func (va *ValidationAuthorityImpl) extractRequestTarget(req *http.Request) (iden
 	}
 
 	if _, err := iana.ExtractSuffix(reqHost); err != nil {
-		return identifier.ACMEIdentifier{}, 0, berrors.ConnectionFailureError("Invalid host in redirect target, must end in IANA registered TLD")
+		isLabca := false
+		for _, domain := range va.labcaDomains {
+			if strings.HasSuffix(reqHost, "."+domain) {
+				isLabca = true
+			}
+		}
+
+		if !isLabca {
+			return identifier.ACMEIdentifier{}, 0, berrors.ConnectionFailureError("Invalid host in redirect target, must end in IANA registered TLD")
+		}
 	}
 
 	return identifier.NewDNS(reqHost), reqPort, nil
