diff --git a/va/va.go b/va/va.go
index fcd35a5db..ad2b7c409 100644
--- a/va/va.go
+++ b/va/va.go
@@ -217,6 +217,7 @@ type ValidationAuthorityImpl struct {
 	perspective        string
 	rir                string
 	isReservedIPFunc   func(ip net.IP) bool
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -237,6 +238,7 @@ func NewValidationAuthorityImpl(
 	perspective string,
 	rir string,
 	reservedIPChecker func(ip net.IP) bool,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -274,6 +276,7 @@ func NewValidationAuthorityImpl(
 		perspective:       perspective,
 		rir:               rir,
 		isReservedIPFunc:  reservedIPChecker,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
