diff --git a/va/va.go b/va/va.go
index 85fa0a63f..d28ae3708 100644
--- a/va/va.go
+++ b/va/va.go
@@ -205,6 +205,7 @@ type ValidationAuthorityImpl struct {
 	rir                  string
 	isReservedIPFunc     func(netip.Addr) error
 	allowRestrictedAddrs bool
+	labcaDomains         []string
 
 	metrics *vaMetrics
 }
@@ -225,6 +226,7 @@ func NewValidationAuthorityImpl(
 	perspective string,
 	rir string,
 	reservedIPChecker func(netip.Addr) error,
+	labcaDomains []string,
 	slowRemoteTimeout time.Duration,
 	allowRestrictedAddrs bool,
 ) (*ValidationAuthorityImpl, error) {
@@ -266,6 +268,7 @@ func NewValidationAuthorityImpl(
 		rir:                  rir,
 		isReservedIPFunc:     reservedIPChecker,
 		allowRestrictedAddrs: allowRestrictedAddrs,
+		labcaDomains:         labcaDomains,
 	}
 
 	return va, nil
