diff --git a/va/va.go b/va/va.go
index 17c03cf6e..237d82c6b 100644
--- a/va/va.go
+++ b/va/va.go
@@ -260,6 +260,7 @@ type ValidationAuthorityImpl struct {
 	singleDialTimeout  time.Duration
 	perspective        string
 	rir                string
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -280,6 +281,7 @@ func NewValidationAuthorityImpl(
 	accountURIPrefixes []string,
 	perspective string,
 	rir string,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -308,6 +310,7 @@ func NewValidationAuthorityImpl(
 		singleDialTimeout: 10 * time.Second,
 		perspective:       perspective,
 		rir:               rir,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
