diff --git a/va/va.go b/va/va.go
index a1e2cd449..883298092 100644
--- a/va/va.go
+++ b/va/va.go
@@ -215,6 +215,7 @@ type ValidationAuthorityImpl struct {
 	singleDialTimeout  time.Duration
 	perspective        string
 	rir                string
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -234,6 +235,7 @@ func NewValidationAuthorityImpl(
 	accountURIPrefixes []string,
 	perspective string,
 	rir string,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -271,6 +273,7 @@ func NewValidationAuthorityImpl(
 		singleDialTimeout: 10 * time.Second,
 		perspective:       perspective,
 		rir:               rir,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
