diff --git a/va/va.go b/va/va.go
index dd743b593..b74a313f0 100644
--- a/va/va.go
+++ b/va/va.go
@@ -265,6 +265,7 @@ type ValidationAuthorityImpl struct {
 	maxRemoteFailures  int
 	accountURIPrefixes []string
 	singleDialTimeout  time.Duration
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -280,6 +281,7 @@ func NewValidationAuthorityImpl(
 	clk clock.Clock,
 	logger blog.Logger,
 	accountURIPrefixes []string,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -306,6 +308,7 @@ func NewValidationAuthorityImpl(
 		// used for the DialContext operations that take place during an
 		// HTTP-01 challenge validation.
 		singleDialTimeout: 10 * time.Second,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
