diff --git a/va/va.go b/va/va.go
index 4993aec36..161cbacda 100644
--- a/va/va.go
+++ b/va/va.go
@@ -219,6 +219,7 @@ type ValidationAuthorityImpl struct {
 	perspective        string
 	rir                string
 	isReservedIPFunc   func(netip.Addr) error
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -239,6 +240,7 @@ func NewValidationAuthorityImpl(
 	perspective string,
 	rir string,
 	reservedIPChecker func(netip.Addr) error,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -276,6 +278,7 @@ func NewValidationAuthorityImpl(
 		perspective:       perspective,
 		rir:               rir,
 		isReservedIPFunc:  reservedIPChecker,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
