diff --git a/va/va.go b/va/va.go
index 270e9ca66..f8585c9fd 100644
--- a/va/va.go
+++ b/va/va.go
@@ -218,6 +218,7 @@ type ValidationAuthorityImpl struct {
 	perspective        string
 	rir                string
 	isReservedIPFunc   func(ip net.IP) bool
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -238,6 +239,7 @@ func NewValidationAuthorityImpl(
 	perspective string,
 	rir string,
 	reservedIPChecker func(ip net.IP) bool,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -275,6 +277,7 @@ func NewValidationAuthorityImpl(
 		perspective:       perspective,
 		rir:               rir,
 		isReservedIPFunc:  reservedIPChecker,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
