diff --git a/va/va.go b/va/va.go
index 55211e27b..20c5f0a94 100644
--- a/va/va.go
+++ b/va/va.go
@@ -220,6 +220,7 @@ type ValidationAuthorityImpl struct {
 	perspective        string
 	rir                string
 	isReservedIPFunc   func(netip.Addr) error
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -240,6 +241,7 @@ func NewValidationAuthorityImpl(
 	perspective string,
 	rir string,
 	reservedIPChecker func(netip.Addr) error,
+	labcaDomains []string,
 	slowRemoteTimeout time.Duration,
 ) (*ValidationAuthorityImpl, error) {
 
@@ -278,6 +280,7 @@ func NewValidationAuthorityImpl(
 		perspective:       perspective,
 		rir:               rir,
 		isReservedIPFunc:  reservedIPChecker,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
