diff --git a/va/va.go b/va/va.go
index d43346bbc..e0784adcb 100644
--- a/va/va.go
+++ b/va/va.go
@@ -256,6 +256,7 @@ type ValidationAuthorityImpl struct {
 	maxRemoteFailures  int
 	accountURIPrefixes []string
 	singleDialTimeout  time.Duration
+	labcaDomains       []string
 
 	metrics *vaMetrics
 }
@@ -274,6 +275,7 @@ func NewValidationAuthorityImpl(
 	clk clock.Clock,
 	logger blog.Logger,
 	accountURIPrefixes []string,
+	labcaDomains []string,
 ) (*ValidationAuthorityImpl, error) {
 
 	if len(accountURIPrefixes) == 0 {
@@ -300,6 +302,7 @@ func NewValidationAuthorityImpl(
 		// used for the DialContext operations that take place during an
 		// HTTP-01 challenge validation.
 		singleDialTimeout: 10 * time.Second,
+		labcaDomains:      labcaDomains,
 	}
 
 	return va, nil
