diff --git a/va/va.go b/va/va.go
index 88f13dc5e..5c48a1f54 100644
--- a/va/va.go
+++ b/va/va.go
@@ -215,6 +215,7 @@ type ValidationAuthorityImpl struct {
 	rir                      string
 	isReservedIPFunc         func(netip.Addr) error
 	allowRestrictedAddrs     bool
+	labcaDomains             []string
 	experimentalVA           *ValidationAuthorityImpl
 	experimentalVASampleRate float64
 	experimentalVATimeout    time.Duration
@@ -238,6 +239,7 @@ func NewValidationAuthorityImpl(
 	perspective string,
 	rir string,
 	reservedIPChecker func(netip.Addr) error,
+	labcaDomains []string,
 	slowRemoteTimeout time.Duration,
 	allowRestrictedAddrs bool,
 	experimentalVA *ValidationAuthorityImpl,
@@ -290,6 +292,7 @@ func NewValidationAuthorityImpl(
 		rir:                      rir,
 		isReservedIPFunc:         reservedIPChecker,
 		allowRestrictedAddrs:     allowRestrictedAddrs,
+		labcaDomains:             labcaDomains,
 		experimentalVA:           experimentalVA,
 		experimentalVASampleRate: experimentalVASampleRate,
 		experimentalVATimeout:    experimentalVATimeout,
