diff --git a/wfe2/wfe.go b/wfe2/wfe.go
index d15a661e8..e45d5789a 100644
--- a/wfe2/wfe.go
+++ b/wfe2/wfe.go
@@ -171,6 +171,8 @@ type WebFrontEndImpl struct {
 	// descriptions (perhaps including URLs) of those profiles. NewOrder
 	// Requests with a profile name not present in this map will be rejected.
 	certProfiles map[string]string
+
+	pa *policy.AuthorityImpl
 }
 
 // NewWebFrontEndImpl constructs a web service for Boulder
@@ -198,6 +200,7 @@ func NewWebFrontEndImpl(
 	unpauseSigner unpause.JWTSigner,
 	unpauseJWTLifetime time.Duration,
 	unpauseURL string,
+	pa *policy.AuthorityImpl,
 ) (WebFrontEndImpl, error) {
 	if len(issuerCertificates) == 0 {
 		return WebFrontEndImpl{}, errors.New("must provide at least one issuer certificate")
@@ -215,6 +218,10 @@ func NewWebFrontEndImpl(
 		return WebFrontEndImpl{}, errors.New("must provide a service for nonce redemption")
 	}
 
+	if pa == nil {
+		return WebFrontEndImpl{}, errors.New("must provide a policy authority")
+	}
+
 	wfe := WebFrontEndImpl{
 		log:                          logger,
 		clk:                          clk,
@@ -239,6 +246,7 @@ func NewWebFrontEndImpl(
 		unpauseSigner:                unpauseSigner,
 		unpauseJWTLifetime:           unpauseJWTLifetime,
 		unpauseURL:                   unpauseURL,
+		pa:                           pa,
 	}
 
 	return wfe, nil
@@ -2274,7 +2282,7 @@ func (wfe *WebFrontEndImpl) NewOrder(
 	}
 
 	names = core.UniqueLowerNames(names)
-	err = policy.WellFormedDomainNames(names)
+	err = wfe.pa.WellFormedDomainNames(names)
 	if err != nil {
 		wfe.sendError(response, logEvent, web.ProblemDetailsForError(err, "Invalid identifiers requested"), nil)
 		return
