diff --git a/wfe2/wfe.go b/wfe2/wfe.go
index 7c12ad7e6..b71d5e2c2 100644
--- a/wfe2/wfe.go
+++ b/wfe2/wfe.go
@@ -177,6 +177,8 @@ type WebFrontEndImpl struct {
 	// descriptions (perhaps including URLs) of those profiles. NewOrder
 	// Requests with a profile name not present in this map will be rejected.
 	certProfiles map[string]string
+
+	pa *policy.AuthorityImpl
 }
 
 // NewWebFrontEndImpl constructs a web service for Boulder
@@ -204,6 +206,7 @@ func NewWebFrontEndImpl(
 	unpauseJWTLifetime time.Duration,
 	unpauseURL string,
 	blockedOnDemandLabels []string,
+	pa *policy.AuthorityImpl,
 ) (WebFrontEndImpl, error) {
 	if len(issuerCertificates) == 0 {
 		return WebFrontEndImpl{}, errors.New("must provide at least one issuer certificate")
@@ -226,6 +229,10 @@ func NewWebFrontEndImpl(
 		blockedLabels = append(blockedLabels, strings.ToLower(label))
 	}
 
+	if pa == nil {
+		return WebFrontEndImpl{}, errors.New("must provide a policy authority")
+	}
+
 	wfe := WebFrontEndImpl{
 		log:                   logger,
 		clk:                   clk,
@@ -250,6 +257,7 @@ func NewWebFrontEndImpl(
 		unpauseJWTLifetime:    unpauseJWTLifetime,
 		unpauseURL:            unpauseURL,
 		blockedOnDemandLabels: blockedLabels,
+		pa:                    pa,
 	}
 
 	return wfe, nil
@@ -692,7 +700,7 @@ func (wfe *WebFrontEndImpl) contactsToEmails(contacts []string) ([]string, error
 			return nil, berrors.InvalidEmailError("contact email contains non-ASCII characters")
 		}
 
-		err = policy.ValidEmail(parsed.Opaque)
+		err = wfe.pa.ValidEmail(parsed.Opaque)
 		if err != nil {
 			return nil, err
 		}
@@ -2348,7 +2356,7 @@ func (wfe *WebFrontEndImpl) NewOrder(
 	idents = identifier.Normalize(idents)
 	logEvent.Identifiers = idents
 
-	err = policy.WellFormedIdentifiers(idents)
+	err = wfe.pa.WellFormedIdentifiers(idents)
 	if err != nil {
 		wfe.sendError(response, logEvent, web.ProblemDetailsForError(err, "Invalid identifiers requested"), nil)
 		return
