diff --git a/cmd/boulder-wfe2/main.go b/cmd/boulder-wfe2/main.go
index 90ad22417..ad57a1ae3 100644
--- a/cmd/boulder-wfe2/main.go
+++ b/cmd/boulder-wfe2/main.go
@@ -92,7 +92,7 @@ type Config struct {
 		// DirectoryCAAIdentity is used for the /directory response's "meta"
 		// element's "caaIdentities" field. It should match the VA's "issuerDomain"
 		// configuration value (this value is the one used to enforce CAA)
-		DirectoryCAAIdentity string `validate:"required,fqdn"`
+		DirectoryCAAIdentity string `validate:"required"`
 		// DirectoryWebsite is used for the /directory response's "meta" element's
 		// "website" field.
 		DirectoryWebsite string `validate:"required,url"`
@@ -160,6 +160,8 @@ type Config struct {
 		// list will be rejected. This field is optional; if unset, no profile
 		// names are accepted.
 		CertificateProfileNames []string `validate:"omitempty,dive,alphanum,min=1,max=32"`
+
+		cmd.HostnamePolicyConfig
 	}
 
 	Syslog        cmd.SyslogConfig
@@ -356,6 +358,7 @@ func main() {
 		txnBuilder,
 		maxNames,
 		c.WFE.CertificateProfileNames,
+		c.WFE.HostnamePolicyFile,
 	)
 	cmd.FailOnError(err, "Unable to create WFE")
 
