diff --git a/wfe2/wfe.go b/wfe2/wfe.go
index aaaf49417..200c76e1d 100644
--- a/wfe2/wfe.go
+++ b/wfe2/wfe.go
@@ -183,6 +183,8 @@ type WebFrontEndImpl struct {
 	// descriptions (perhaps including URLs) of those profiles. NewOrder
 	// Requests with a profile name not present in this map will be rejected.
 	certProfiles map[string]string
+
+	pa *policy.AuthorityImpl
 }
 
 // NewWebFrontEndImpl constructs a web service for Boulder
@@ -210,6 +212,7 @@ func NewWebFrontEndImpl(
 	unpauseJWTLifetime time.Duration,
 	unpauseURL string,
 	blockedOnDemandLabels []string,
+	pa *policy.AuthorityImpl,
 	caaIdentity string,
 ) (WebFrontEndImpl, error) {
 	if len(issuerCertificates) == 0 {
@@ -238,6 +241,10 @@ func NewWebFrontEndImpl(
 		return WebFrontEndImpl{}, fmt.Errorf("normalizing caaIdentity: %w", err)
 	}
 
+	if pa == nil {
+		return WebFrontEndImpl{}, errors.New("must provide a policy authority")
+	}
+
 	wfe := WebFrontEndImpl{
 		log:                   logger,
 		clk:                   clk,
@@ -262,6 +269,7 @@ func NewWebFrontEndImpl(
 		unpauseJWTLifetime:    unpauseJWTLifetime,
 		unpauseURL:            unpauseURL,
 		blockedOnDemandLabels: blockedLabels,
+		pa:                    pa,
 		DirectoryCAAIdentity:  normalizedCAAIdentity,
 	}
 
@@ -705,7 +713,7 @@ func (wfe *WebFrontEndImpl) contactsToEmails(contacts []string) ([]string, error
 			return nil, berrors.InvalidEmailError("contact email contains non-ASCII characters")
 		}
 
-		err = policy.ValidEmail(parsed.Opaque)
+		err = wfe.pa.ValidEmail(parsed.Opaque)
 		if err != nil {
 			return nil, err
 		}
@@ -2384,7 +2392,7 @@ func (wfe *WebFrontEndImpl) NewOrder(
 	idents = identifier.Normalize(idents)
 	logEvent.Identifiers = idents
 
-	err = policy.WellFormedIdentifiers(idents)
+	err = wfe.pa.WellFormedIdentifiers(idents)
 	if err != nil {
 		wfe.sendError(response, logEvent, web.ProblemDetailsForError(err, "Invalid identifiers requested"), nil)
 		return
