diff --git a/cmd/bad-key-revoker/main.go b/cmd/bad-key-revoker/main.go
index 17704c41..e2a060b1 100644
--- a/cmd/bad-key-revoker/main.go
+++ b/cmd/bad-key-revoker/main.go
@@ -14,6 +14,7 @@ import (
 	"time"
 
 	"github.com/honeycombio/beeline-go"
+	"github.com/letsencrypt/boulder/bdns"
 	"github.com/letsencrypt/boulder/cmd"
 	"github.com/letsencrypt/boulder/core"
 	corepb "github.com/letsencrypt/boulder/core/proto"
@@ -346,6 +347,9 @@ func main() {
 			TLS       cmd.TLSConfig
 			RAService *cmd.GRPCClientConfig
 
+			DNSTries     int
+			DNSResolvers []string
+
 			// MaximumRevocations specifies the maximum number of certificates associated with
 			// a key hash that bad-key-revoker will attempt to revoke. If the number of certificates
 			// is higher than MaximumRevocations bad-key-revoker will error out and refuse to
@@ -373,6 +377,12 @@ func main() {
 
 		Syslog  cmd.SyslogConfig
 		Beeline cmd.BeelineConfig
+
+		Common struct {
+			DNSResolver               string
+			DNSTimeout                string
+			DNSAllowLoopbackAddresses bool
+		}
 	}
 	configPath := flag.String("config", "", "File path to the configuration file for this service")
 	flag.Parse()
@@ -418,6 +428,30 @@ func main() {
 	cmd.FailOnError(err, "Failed to load credentials and create gRPC connection to RA")
 	rac := rapb.NewRegistrationAuthorityClient(conn)
 
+	dnsTimeout, err := time.ParseDuration(config.Common.DNSTimeout)
+	cmd.FailOnError(err, "Couldn't parse DNS timeout")
+	dnsTries := config.BadKeyRevoker.DNSTries
+	if dnsTries < 1 {
+		dnsTries = 1
+	}
+	var resolver bdns.Client
+	if len(config.Common.DNSResolver) != 0 {
+		config.BadKeyRevoker.DNSResolvers = append(config.BadKeyRevoker.DNSResolvers, config.Common.DNSResolver)
+	}
+	if !config.Common.DNSAllowLoopbackAddresses {
+		r := bdns.New(
+			dnsTimeout,
+			bdns.NewStaticProvider(config.BadKeyRevoker.DNSResolvers),
+			scope,
+			clk,
+			dnsTries,
+			logger)
+		resolver = r
+	} else {
+		r := bdns.NewTest(dnsTimeout, bdns.NewStaticProvider(config.BadKeyRevoker.DNSResolvers), scope, clk, dnsTries, logger)
+		resolver = r
+	}
+
 	var smtpRoots *x509.CertPool
 	if config.BadKeyRevoker.Mailer.SMTPTrustedRootFile != "" {
 		pem, err := ioutil.ReadFile(config.BadKeyRevoker.Mailer.SMTPTrustedRootFile)
@@ -439,6 +473,7 @@ func main() {
 		config.BadKeyRevoker.Mailer.Username,
 		smtpPassword,
 		smtpRoots,
+		resolver,
 		*fromAddress,
 		logger,
 		scope,
