diff --git a/core/interfaces.go b/core/interfaces.go
index c57358c9b..5c5087c2f 100644
--- a/core/interfaces.go
+++ b/core/interfaces.go
@@ -108,6 +108,7 @@ type PolicyAuthority interface {
 	WillingToIssueWildcards(identifiers []identifier.ACMEIdentifier) error
 	ChallengesFor(domain identifier.ACMEIdentifier) ([]Challenge, error)
 	ChallengeTypeEnabled(t string) bool
+	ValidEmail(address string) error
 }
 
 // StorageGetter are the Boulder SA's read-only methods
