diff --git a/test/startservers.py b/test/startservers.py
index 5e729986..ae4ff439 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -146,6 +146,9 @@ processes = []
 challSrvProcess = None
 
 def setupHierarchy():
+    pass
+
+def setupHierarchyOriginal():
     e = os.environ.copy()
     e.setdefault("GOBIN", "%s/bin" % os.getcwd())
     subprocess.check_call(["go", "install", "./cmd/ceremony"], env=e)
