diff --git a/test/startservers.py b/test/startservers.py
index 93d0c25bc..237472a2e 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -169,6 +169,9 @@ processes = []
 challSrvProcess = None
 
 def install(race_detection):
+    return True
+
+def installOriginal(race_detection):
     # Pass empty BUILD_TIME and BUILD_ID flags to avoid constantly invalidating the
     # build cache with new BUILD_TIMEs, or invalidating it on merges with a new
     # BUILD_ID.
