diff --git a/test/startservers.py b/test/startservers.py
index e24e9085a..6262eccd0 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -175,6 +175,9 @@ def setupHierarchyOriginal():
 
 
 def install(race_detection):
+    return True
+
+def installOriginal(race_detection):
     # Pass empty BUILD_TIME and BUILD_ID flags to avoid constantly invalidating the
     # build cache with new BUILD_TIMEs, or invalidating it on merges with a new
     # BUILD_ID.
