diff --git a/test/startservers.py b/test/startservers.py
index df82abbf8..08720c37e 100644
--- a/test/startservers.py
+++ b/test/startservers.py
@@ -186,6 +186,9 @@ processes = []
 challSrvProcess = None
 
 def install(race_detection, coverage=False):
+    return True
+
+def installOriginal(race_detection):
     # Pass empty BUILD_TIME and BUILD_ID flags to avoid constantly invalidating the
     # build cache with new BUILD_TIMEs, or invalidating it on merges with a new
     # BUILD_ID.
